### ISSUE TITLE
SC2078 doesn't expect command in expression

### ISSUE THREAD

**Issue by KawaiDesu** [koalaman/shellcheck#1054]
- Rule ID: SC2078
- version: 0.4.6
- [✓] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [✕] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet that shows the problem:
```sh
#!/bin/bash

is_valid(){
    # exits with code 1
    grep -q ololo /proc/cpuinfo
}

if [ ! is_valid ]; then
    echo "ok"
fi
```

#### Here's what shellcheck currently says:
![_2017-11-16_21-08-20](https://user-images.githubusercontent.com/5793027/32907778-4c156f72-cb12-11e7-96fc-0cfdf3cbd5e6.png)

#### Here's what I wanted or expected to see:
Nothing

#### Why
Looks like shellsheck does not understand command in test expression. It is valid as soon as there is no parameters to command, like "true" or "false". But unlike true/false my command result needs to be inverted in that place, so i can't just remove [brackets]

**Comment by koalaman** [comment]
ShellCheck is right. This is a constant expression. You do need to remove the brackets.

    if ! is_valid
    then
      echo "invalid"
    fi

**Comment by KawaiDesu** [comment]
If so, this must be SC2160
> Instead of '[ true ]', just use 'true'.

with modified text, not SC2078.

**Comment by koalaman** [comment]
That would indeed have been better in this case, and ShellCheck might have been able to figure it out based on the literal string matching a defined function.

In the general case though, ` [ ! alphanumeric ]` is more likely supposed to be `[ ! $alphanumeric ]` than `! alphanumeric`, which is why the message is what it currently is.
